gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/importExport/GradeImportConfirmationStep.java

```diff
@@ -203,7 +203,8 @@ public void onSubmit(AjaxRequestTarget target, Form<?> form) {
 				// Modify any that need modification
 				itemsToModify.forEach(item -> {
 
-					final Double points = NumberUtils.toDouble(item.getItemPointValue());
+					final String itemPoints = StringUtils.replace(item.getItemPointValue(), ",", ".");
+					final Double points = NumberUtils.toDouble(itemPoints);
```

Comment:
replace commas with periods just seems so wrong ..... no?!

why don't we have a centralized method to do this by now? why aren't there unit tests?

Reply:
Unfortunately this happens in many places in GBNG

Reply:
you're okay with your instructor commas being replaced with periods?!

  private static String formatGradeForLocale(final String grade, final Locale locale) {
    if (StringUtils.isBlank(grade)) {
      return "";
    }

    String s;
    try {
      final DecimalFormat df = (DecimalFormat) NumberFormat.getInstance(locale);
      final Double d = df.parse(grade).doubleValue();

      df.setMinimumFractionDigits(0);
      df.setGroupingUsed(false);

      s = df.format(d);
    } catch (final NumberFormatException | ParseException e) {
      log.warn("Bad format, returning original string: {}", grade);
      s = grade;
    }

    return StringUtils.removeEnd(s, ".0");
  }

Reply:
This is in the context of a grade cell in a spreadsheet. Previously in a decimal locale, if a comma was present this code would explode. So, yes, this is arguable, but marginally better than before.

If you would rather see a completely new centralized approach with unit tests, I'm fine with that as well. I just didn't think this was the JIRA to do that for.

Reply:
I agree that this is probably not the best way to do it, but I found several places where this replacement method is used to do the same. We should probably work on a centralized method to do this.

Reply:
Sorry but this just introduces a new bug for default (USA) locale!

1. Create a new GB item worth 1000 points
2. Export to CSV
3. Modify CSV by hand
4. Note the header and the comma! "Student ID","Name","# Student Number","New Item [1,000]"," * New Item"
5. Modify one character and make it worth 1,001 points
6. Import

Reply:
gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/util/FormatHelper.java seems like the right place.... or you could go to kernel .... knowing the locale seems important before doing a replace

There are tests here:

gradebookng/tool/src/test/org/sakaiproject/gradebookng/business/TestGradebookNgBusinessService.java

Reply:
Hi again, I have fixed the PR using the FormatHelper class